src/Rules/Node/ValidConstantFunctionRule.php

```diff
@@ -26,27 +26,39 @@ public function enterNode(Node $node, Environment $env): Node
         }
 
         $arguments = $node->getNode('arguments');
-        if (1 !== $arguments->count()) {
-            return $node;
+        $argument = $arguments->hasNode('0') ? $arguments->getNode('0') : null;
+        // Try for named parameters
+        if (null === $argument && $arguments->hasNode('constant')) {
+            $argument = $arguments->getNode('constant');
         }
-
-        $argument = $arguments->getNode('0');
         if (!$argument instanceof ConstantExpression) {
             return $node;
         }
 
         $constant = $argument->getAttribute('value');
         if (!\is_string($constant)) {
+            $this->addError(
+                'The first param of the function "constant()" must be a string.',
+                $node,
+                'StringConstant'
+            );
+
+            return $node;
+        }
+
+        // The object to get the constant from cannot be resolved statically.
+        if (1 !== $arguments->count()) {
             return $node;
         }
 
         if (\defined($constant)) {
             return $node;
         }
 
+
         if ('::class' === strtolower(substr($constant, -7))) {
             $this->addError(
-                sprintf('You cannot use the Twig function "constant()" to access "%s". You could provide an object and call constant("class", $object) or use the class name directly as a string.', $constant),
```

Comment:
I'll take any suggestion here @ruudk but I'd like to avoid a too long error message.

So far I always had short message like "Variable should be called foo" or "Expected 2 spaces, found 1". This gave report with all the error messages written on single line every time and I'm not happy with the fact that this error would be certainly written with multiples lines.

Maybe letter something like an additional help text could be written, but so far I'd like a short message.
I try `'You cannot use the function "constant()" to access class names.'` but take any suggestion

Reply:
Fine. This is coming directly from https://github.com/twigphp/Twig/blob/1377f2da4efa020d2ad9b8fb8b722d688ab297ad/src/Extension/CoreExtension.php#L1474